src/System.Diagnostics.PerformanceCounter/src/System/Diagnostics/PerformanceCounterLib.cs

```diff
@@ -1636,6 +1655,24 @@ internal InstanceDataCollectionCollection ReadCategory()
 
             return data;
         }
+
+        public void Dispose()
+        {
+            if (Interlocked.Exchange(ref _disposed, 1) == 1)
```

Comment:
Why the Interlocked.Exchange?  If we're concerned about concurrent misuse, then we should similarly be concerned about other members (like GetInstanceNamesFromIndex and GetCounterDefinitionSample) being used concurrently with Dispose... this Exchange won't help with the race conditions there, even with the CheckDisposed calls.

Reply:
Removed. Using simple bool. Thanks!